src/config.ts

```diff
@@ -138,14 +138,20 @@ export class Config {
   }
 
   // Return the config with an overwritten seed URL.
-  withSeed(seedHost: string, seedId?: string): Config {
+  withSeed(seed: Seed): Config {
     const cfg = {} as Config;
     Object.assign(cfg, this);
-    cfg.seed.host = seedHost;
 
-    if (seedId) {
-      cfg.seed.id = seedId;
-    }
+    // The `git` and `api` keys being more specific take
+    // precedence over the `host`, if available.
+    const api = seed.api || seed.host;
+    const git = seed.git || seed.host;
```

Comment:
I think this is a good place to use the nullish coalescing operator, since we are checking for nullish values.

```suggestion
    const api = seed.api ?? seed.host;
    const git = seed.git ?? seed.host;
```